source/postprocess/visualization/maximum_horizontal_compressive_stress.cc

```diff
@@ -47,134 +47,161 @@ namespace aspect
+        Assert ((computed_quantities[0].size() == dim), ExcInternalError());
         Assert (input_data.solution_values[0].size() == this->introspection().n_components,   ExcInternalError());
         Assert (input_data.solution_gradients[0].size() == this->introspection().n_components,  ExcInternalError());
-        AssertThrow (this->get_parameters().enable_elasticity == false, ExcMessage("The maximum horizontal compressive stress plugin currently does not work with elasticity."));
 
         MaterialModel::MaterialModelInputs<dim> in(input_data,
                                                    this->introspection());
         MaterialModel::MaterialModelOutputs<dim> out(n_quadrature_points,
                                                      this->n_compositional_fields());
 
-        // We do not need to compute anything but the viscosity
-        in.requested_properties = MaterialModel::MaterialProperties::viscosity;
+        if (this->get_parameters().enable_elasticity)
+          in.requested_properties = MaterialModel::MaterialProperties::viscosity
+                                    | MaterialModel::MaterialProperties::additional_outputs;
+        else
+          in.requested_properties = MaterialModel::MaterialProperties::viscosity;
+
+        if (in.requested_properties & MaterialModel::MaterialProperties::additional_outputs)
+          this->get_material_model().create_additional_named_outputs(out);
 
-        // Compute the viscosity...
         this->get_material_model().evaluate(in, out);
 
-        // ...and use it to compute the stresses and from that the
-        // maximum compressive stress direction
+        // helper maps + lambdas to pack/unpack symmetric tensor components
+        std::vector<std::pair<unsigned,unsigned>> comp_map;
+        if (dim==2)
+          comp_map = {{0u,0u},{1u,1u},{0u,1u}};
+        else
+          comp_map = {{0u,0u},{1u,1u},{2u,2u},{0u,1u},{0u,2u},{1u,2u}};
+
+        auto pack_tensor = [&](const SymmetricTensor<2,dim> &t,
+                               Vector<double> &v)
+        {
+          Assert (v.size() == comp_map.size(), ExcInternalError());
+          for (unsigned k=0; k<v.size(); ++k)
+            {
+              const auto &p = comp_map[k];
+              v[k] = t[p.first][p.second];
+            }
+        };
+
+        auto unpack_tensor = [&](const Vector<double> &v,
+                                 SymmetricTensor<2,dim> &t)
+        {
+          Assert (v.size() == comp_map.size(), ExcInternalError());
+          t = SymmetricTensor<2,dim>(); // zero
+          for (unsigned k=0; k<v.size(); ++k)
+            {
+              const auto &p = comp_map[k];
+              t[p.first][p.second] = v[k];
+              if (p.first != p.second)
+                t[p.second][p.first] = v[k];
+            }
+        };
+
+        // --- STEP 1: compute compressive stress components at quadrature points
+        const unsigned int n_tensor_components = (dim==2 ? 3u : 6u);
```

Comment:
Not wrong, but this is simpler:
```suggestion
        const unsigned int n_tensor_components = SymmetricTensor<2,dim>::n_independent_components;
```